src/sql/parts/jobManagement/views/jobsView.component.ts

```diff
@@ -283,43 +281,68 @@ export class JobsViewComponent implements AfterContentChecked {
 	}
 
 	private loadJobHistories() {
-		const self = this;
 		if (this.jobs) {
 			let erroredJobs = 0;
-			for (let i = 0; i < this.jobs.length; i++) {
-				let job = this.jobs[i];
-				let ownerUri: string = this._dashboardService.connectionManagementService.connectionInfo.ownerUri;
-				this._jobManagementService.getJobHistory(ownerUri, job.jobId).then((result) => {
-					if (result.jobs) {
-						self.jobHistories[job.jobId] = result.jobs;
-						self._jobCacheObject.setJobHistory(job.jobId, result.jobs);
-						if (self._agentViewComponent.expanded.has(job.jobId)) {
-							let jobHistory = self._jobCacheObject.getJobHistory(job.jobId)[0];
-							let item = self.dataView.getItemById(job.jobId + '.error');
-							let noStepsMessage = nls.localize('jobsView.noSteps', 'No Steps available for this job.');
-							let errorMessage = jobHistory ? jobHistory.message: noStepsMessage;
-							item['name'] = nls.localize('jobsView.error', 'Error: ') + errorMessage;
-							self._agentViewComponent.setExpanded(job.jobId, errorMessage);
-							self.dataView.updateItem(job.jobId + '.error', item);
-
-						}
-					}
-				});
+			let ownerUri: string = this._dashboardService.connectionManagementService.connectionInfo.ownerUri;
+			let separatedJobs = this.separateFailingJobs();
+			// grab histories of the failing jobs first
+			// so they can be expanded quicker
+			let failing = separatedJobs[0];
+			this.curateJobHistory(failing, ownerUri);
+			let passing = separatedJobs[1];
+			this.curateJobHistory(passing, ownerUri);
+		}
+	}
+
+	private separateFailingJobs(): any {
```

Comment:
any reason to use `any` here?  looks like an array of array of AgentJobInfo, or something like that?